rclpy/rclpy/__init__.py

```diff
@@ -12,12 +12,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import rclpy._rclpy__rmw_opensplice_cpp as _rclpy
+import importlib
+import os
+
+import ament_index_python
 
 from rclpy.node import Node
 
+rmw_implementations = sorted(ament_index_python.get_resources('rmw_implementation').keys())
+
+
+def _load_rmw(rmw_implementation):
+    module_name = '._rclpy__{rmw_implementation}'.format(
+        rmw_implementation=rmw_implementation,
+    )
+    return importlib.import_module(module_name, package='rclpy')
+
+
+global _rclpy
+try:
+    _rclpy
+except NameError:
+    _rclpy = None
+if _rclpy is None:
+    for rmw_implementation in rmw_implementations:
+        rmw_module = _load_rmw(rmw_implementation)
+        if rmw_module is not None:
+            _rclpy = rmw_module
+            break
+
+
+def set_rmw_implementation(rmw_implementation):
```

Comment:
@wjwwood I've implemented the approach you describe and triggered the CI, I think I haven't missed anything, but could you have a look at it? Thanks!